src/structures/Download.js

```diff
@@ -0,0 +1,37 @@
+const fs = require('fs/promises')
+const fetch = require('node-fetch')
+
+module.exports = class Download {
+  #api
+  constructor (api, data) {
+    this.#api = api
+    this.downloadUrl = data.downloadLink ?? data.downloadUrl
+    if (this.#api.platform === 'bedrock') {
+      this.token = data.token
+      this.size = data.size
+      this.fileExtension = '.mcworld'
+    } else {
+      this.resourcePackUrl = data.resourcePackUrl
+      this.resourcePackHash = data.resourcePackHash
+      this.fileExtension = '.tar.gz'
+    }
+  }
+
+  async toDirectory (directory) {
+    return this.#downloadWorld().then(buffer => fs.writeFile(`${directory}/world${this.fileExtension}`, buffer))
+  }
+
+  async toBuffer () {
```

Comment:
for clarity, rename the methods to `getBuffer()` and `writeToDirectory()`. Also, use fs promises and only return and resolve once the download and write to disk is complete.

Reply:
Will do, it already uses fs promsises and does only return once the downloads done.